compass/Coordinator.java

```diff
@@ -262,6 +280,11 @@ private void start() throws InterruptedException {
         }
         //normal flow
         else {
+          // We want to perform shutdown only when we are ready to issue the next normal milestone
+            // and the node is in sync with the latest milestone we issued.
+          if (shutdown) {
```

Comment:
small caveat - this will not stop the bootstrap stage, so if I run w/ `-bootstrap` it will only stop on the 3rd milestone:

```
2019-04-18 11:02:34:763 +0000 [main] INFO org.iota.compass.Coordinator - Broadcast milestone: 1
2019-04-18 11:02:34:763 +0000 [main] INFO org.iota.compass.Coordinator - Depth: 3 -> 3
2019-04-18 11:02:34:769 +0000 [main] INFO org.iota.compass.Coordinator - stored index 1
2019-04-18 11:02:44:775 +0000 [main] INFO org.iota.compass.Coordinator - Reusing last milestone.
2019-04-18 11:02:44:776 +0000 [main] INFO org.iota.compass.Coordinator - Issuing milestone: 2
2019-04-18 11:02:44:776 +0000 [main] INFO org.iota.compass.Coordinator - Trunk: MXBXNLPXCDNJZQNHBLRBSSSKWMJBCBYHHSHETQXWQFBIXNBEYIXECUEHSUXHNMXFMPUTKPJGYZ9TW9999 Branch: 999999999999999999999999999999999999999999999999999999999999999999999999999999999
2019-04-18 11:02:47:007 +0000 [Shutdown Hook] INFO org.iota.compass.Coordinator - Shutting down Compass after next milestone...
2019-04-18 11:03:00:813 +0000 [main] INFO org.iota.compass.Coordinator - Broadcast milestone: 2
2019-04-18 11:03:00:813 +0000 [main] INFO org.iota.compass.Coordinator - Depth: 3 -> 3
2019-04-18 11:03:00:814 +0000 [main] INFO org.iota.compass.Coordinator - stored index 2
2019-04-18 11:03:10:821 +0000 [main] INFO org.iota.compass.Coordinator - Reusing last milestone.
2019-04-18 11:03:10:821 +0000 [main] INFO org.iota.compass.Coordinator - Issuing milestone: 3
2019-04-18 11:03:10:821 +0000 [main] INFO org.iota.compass.Coordinator - Trunk: AONPB9NU9ICYRVMPMBGCDKIYVXVGQETAXZY9AFLMJFDBDBQLKKAREBRJISCGEPBONRKGYHXOXNMN99999 Branch: 999999999999999999999999999999999999999999999999999999999999999999999999999999999
2019-04-18 11:03:15:305 +0000 [main] INFO org.iota.compass.Coordinator - Broadcast milestone: 3
2019-04-18 11:03:15:305 +0000 [main] INFO org.iota.compass.Coordinator - Timestamp delta: 14491
2019-04-18 11:03:15:305 +0000 [main] INFO org.iota.compass.Coordinator - Depth: 3 -> 3
2019-04-18 11:03:15:306 +0000 [main] INFO org.iota.compass.Coordinator - stored index 3
➜  private_tangle (karimodm-graceful-shutdown|…) % 
```

Reply:
This was made on purpose! I just want it to handle a graceful shutdown when reached a "normal" operation mode. Interrupting bootstrap mid-flight should be explicitly forced.
Do you agree?

Reply:
yes, I agree. just wanted to make sure it was done knowingly.